src/middlewares/accomodation.middleware.js

```diff
@@ -39,5 +39,31 @@ class AccomodationMiddleware {
     }
     return next();
   }
+
+  /** checks if the accomodation selected is available
+   *  @param {req} req it contains the request of the user
+   *  @param {res} res it contains the response the user receive
+   *  @param {function} next it jumps to the next middleware in the route
+   * @return {object} the data from the first middleware
+   */
+  static async checkBookingRequestAvailability(req, res, next) {
+    const departureDate = new Date(req.body.departureDate);
+    const checkoutDate = new Date(req.body.checkoutDate);
+    if (checkoutDate <= departureDate) {
+      return response.errorMessage(res, 'The departure date cannot be later than the return date', 400);
+    }
+    const accomodation = await accommodationService.findAccomodation(req.body.accommodationId);
+    if (!accomodation) {
+      return response.errorMessage(res, 'You have provided an ivalid accommodation', 404);
```

Comment:
@VedaDom Kindly check the spelling of your return messages. Additionally, here you're checking that accommodation exists in the database not if the accommodationID is valid. Therefore the response message is misleading. Kindly update the response message and validate the accommodationID elsewhere.